benchmark/portblas/blas3/gemm_batched.cpp

```diff
@@ -239,13 +239,227 @@ void register_benchmark(blas_benchmark::Args& args,
+            m * n * batch_size, q);
+    auto copy_temp = blas::helper::copy_to_device(
+        q, reinterpret_cast<blas::complex_sycl<scalar_t>*>(c_temp.data()),
+        c_temp_gpu, m * n * batch_size);
+    sb_handle.wait(copy_temp);
+    auto gemm_batched_event = _gemm_batched(
+        sb_handle, *t_a, *t_b, m, n, k, alpha_sycl, a_gpu, lda, b_gpu, ldb,
+        beta_sycl, c_temp_gpu, ldc, batch_size, batch_type);
+    sb_handle.wait(gemm_batched_event);
+    auto copy_out = blas::helper::copy_to_host(
+        q, c_temp_gpu,
+        reinterpret_cast<blas::complex_sycl<scalar_t>*>(c_temp.data()),
+        m * n * batch_size);
+    sb_handle.wait(copy_out);
+
+    blas::helper::deallocate<mem_alloc>(c_temp_gpu, q);
+  }
+
+  std::ostringstream err_stream;
+  if (!utils::compare_vectors<scalar_t>(c_temp, c_ref, err_stream, "")) {
+    const std::string& err_str = err_stream.str();
+    state.SkipWithError(err_str.c_str());
+    *success = false;
+  };
+#endif  // BLAS_VERIFY_BENCHMARK
+
+  auto blas_method_def = [&]() -> std::vector<cl::sycl::event> {
+    auto event = _gemm_batched(sb_handle, *t_a, *t_b, m, n, k, alpha_sycl,
+                               a_gpu, lda, b_gpu, ldb, beta_sycl, c_gpu, ldc,
+                               batch_size, batch_type);
+    sb_handle.wait(event);
+    return event;
+  };
+
+  // Warmup
+  blas_benchmark::utils::warmup(blas_method_def);
+  sb_handle.wait();
+
+  blas_benchmark::utils::init_counters(state);
+
+  // Measure
+  for (auto _ : state) {
+    // Run
+    std::tuple<double, double> times =
+        blas_benchmark::utils::timef(blas_method_def);
+
+    // Report
+    blas_benchmark::utils::update_counters(state, times);
+  }
+
+  state.SetItemsProcessed(state.iterations() * state.counters["n_fl_ops"]);
+  state.SetBytesProcessed(state.iterations() *
+                          state.counters["bytes_processed"]);
+
+  blas_benchmark::utils::calc_avg_counters(state);
+
+  blas::helper::deallocate<mem_alloc>(a_gpu, q);
+  blas::helper::deallocate<mem_alloc>(b_gpu, q);
+  blas::helper::deallocate<mem_alloc>(c_gpu, q);
+};
+
+/*! @brief Register & run benchmark of complex data types gemm batched.
+ * Function is similar to register_benchmark
+ *
+ * @tparam scalar_t element data type of underlying complex (float or double)
+ * @tparam mem_alloc USM or Buffer memory allocation
+ */
+template <typename scalar_t, blas::helper::AllocType mem_alloc>
+void register_cplx_benchmark(
+    blas::SB_Handle* sb_handle_ptr, bool* success, std::string mem_type,
+    std::vector<gemm_batched_cplx_param_t<scalar_t>> params) {
+  for (auto p : params) {
+    std::string t1s, t2s;
+    index_t m, n, k, batch_size;
+    scalar_t alpha_r, alpha_i, beta_r, beta_i;
+    int batch_type;
+    std::tie(t1s, t2s, m, k, n, alpha_r, alpha_i, beta_r, beta_i, batch_size,
+             batch_type) = p;
+    // Only batch_type == strided is supported with Complex data
+    if (batch_type == 1) {
```

Comment:
If we want to keep this check here than we don't really need to check for the `batch_type` in `run()` function since we are manually choosing the type of data layout that is going to be called for the `run()` function.